### PR TITLE
nixos/shadowsocks: add test without plugin

### PR DESCRIPTION
**Pull request by hmenke**
###### Motivation for this change

Fixes https://github.com/NixOS/nixpkgs/pull/96127#issuecomment-688849348

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/networking/shadowsocks.nix [hunk 1]
```diff
@@ -129,7 +147,7 @@ in
       description = "shadowsocks-libev Daemon";
       after = [ "network.target" ];
       wantedBy = [ "multi-user.target" ];
-      path = [ pkgs.shadowsocks-libev cfg.plugin ] ++ optional (cfg.passwordFile != null) pkgs.jq;
+      path = [ pkgs.shadowsocks-libev ] ++ optional (cfg.plugin != null) cfg.plugin ++ optional (cfg.passwordFile != null) pkgs.jq;
```
Comment (hmenke) [inline nixos/modules/services/networking/shadowsocks.nix:150]: This should fix https://github.com/NixOS/nixpkgs/pull/96127#issuecomment-688849348

**Comment by hmenke** [comment]
Result of `nixpkgs-review pr 97478` [1](https://github.com/Mic92/nixpkgs-review)

File: nixos/modules/services/networking/shadowsocks.nix (hunk 1 above)
Comment (Mrmaxmeier) [reply nixos/modules/services/networking/shadowsocks.nix:150]: LGTM :+1:

**Comment by hmenke** [comment]
@ofborg test shadowsocks.shadowsocks shadowsocks.shadowsocks-v2ray-plugin

**Comment by hmenke** [comment]
@ofborg test shadowsocks.basic shadowsocks.v2ray-plugin

**Comment by hmenke** [comment]
@bennofs I see you backported it to 20.09 as well already.  Thank you very much!

**Pull request merged**